nixos/modules/services/misc/octoprint.nix

```diff
@@ -17,9 +17,7 @@ let
 
   cfgUpdate = pkgs.writeText "octoprint-config.yaml" (builtins.toJSON fullConfig);
 
-  pluginsEnv = package.python.withPackages (ps: [ps.octoprint] ++ (cfg.plugins ps));
-
-  package = pkgs.octoprint;
+  pluginsEnv = cfg.package.python.withPackages (ps: [ps.octoprint] ++ (cfg.plugins ps));
```

Comment:
this is probably not the intended affect, as the `package` which gets passed gets replaced by the one which exists in the python3Packages set

Reply:
I see, is there a way around that?

Reply:
Best way would probably to be to create an overlay for the python package set, then build a `python.withPackage` environment with that overlay applied.

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-override-a-python-package-how-to-override-a-python-package is what I'm talking about